content/blog/knative-eventing-for-observability.md

```diff
@@ -0,0 +1,497 @@
+```
+
+
+We can now use any number of consumers, the most simple being the Knative event-display container (`gcr.io/knative-releases/knative.dev/eventing-contrib/cmd/event_display)`.
+
+For this example, I’ve created and published a container using the code in the repo. We’re going to deploy it along with another role that gives it access to the horizontal pod autoscalers, as well as a trigger to send the events to the container:
+
+
+```yaml
+apiVersion: v1
+kind: ServiceAccount
+metadata:
+  name: hpa-sender
+  namespace: knative-eventing
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRole
+metadata:
+  name: hpa-sender
+rules:
+- apiGroups: ["", "autoscaling"] # "" indicates the core API group
+  resources: ["secrets", "horizontalpodautoscalers"]
+  verbs: ["get", "watch", "list"]
+
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRoleBinding
+metadata:
+  name: hpa-sender
+roleRef:
+  apiGroup: rbac.authorization.k8s.io
+  kind: ClusterRole
+  name: hpa-sender
+subjects:
+- kind: ServiceAccount
+  name: hpa-sender
+  namespace: knative-eventing
+---
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: hpa-sender
+  namespace: knative-eventing
+  labels:
+    app: hpa-sender
+spec:
+  replicas: 1
+  selector:
+    matchLabels:
+      app: hpa-sender
+  template:
+    metadata:
+      labels:
+        app: hpa-sender
+    spec:
+      serviceAccountName: hpa-sender
+      containers:
+      - name: hpa-sender
+        image: vmtyler/hpa-sender
+        ports:
+        - containerPort: 8080
+
+---
+apiVersion: v1
+kind: Service
+metadata:
+  name: hpa-sender
+  namespace: knative-eventing
+spec:
+  selector:
+    app: hpa-sender
+  ports:
+    - protocol: TCP
+      port: 80  
+      targetPort: 8080
+---
+apiVersion: eventing.knative.dev/v1
+kind: Trigger
+metadata:
+  name: hpa-sender-trigger
+  namespace: knative-eventing
+spec:
+  broker: default
+  subscriber:
+    uri: http://hpa-sender.knative-eventing.svc.cluster.local
+```
+
+
+We can apply these all together from a file:
+
+
+```bash
+kubectl apply -f hpa-sender-deployment.yaml
+```
+
+
+
+#### HPA-Sender Configuration
+
+The way the hpa-sender container works is pretty simple; it:
+
+
+
+*   Watches the API server events related to HPAs
+*   Retrieves the specific HPA referenced in the event from the API server
+*   Checks if the HPA has an annotation for hpa-sender with a secret location as namespace/secretname
+*   Retrieves the secret, which includes configuration for where to send the event, if it has the annotation
+
+So to use this app, we need a secret with the configuration necessary for our event destination, and we have to annotate our HPA.
+
+
+#### Sample App and HPA Secret
+
+The first thing we need to do is create the secret with the necessary configuration. (There are some sample ones in the repo.) We’re going to use WaveFront as our destination:
```

Comment:
(Maybe worth adding some quick message about what Wavefront is with a link to the guide about it?)